### PR TITLE
Add Actor Name to Check Dialog Title for GMs

### LINKED ISSUE #1384: Request: Actor in name of Roll modifier window
In GitLab by @f.pomerleau0 on May 5, 2021, 13:29

Currently when rolling a roll with the complete roll modifier window, 

image

there's no way of knowing for which actor it's rolling. I would be a nice QoL upgrade when rolling batch rolls to have the name of the token appears in the header. For exemple, it could Say : NAME: Saving throw: Reflex

### PR DESCRIPTION
**Pull request by DocSchlock**
Closes #1384

I limited the name appearing in the title to GMs only, as utility to players is slim.

### CODE REVIEW

File: src/module/system/check/dialog.ts [hunk 1]
```diff
@@ -29,14 +29,18 @@ export class CheckModifiersDialog extends Application {
     ) {
         // The title often has HTML in it: get the base text
         const title = ((): string => {
+            // add actor name to title to differentiate rolls
+            const actorName = context.actor?.name.trim() ?? "";
+
             const maybeWithHTML = context.title?.trim() || check.slug;
-            if (!maybeWithHTML.includes("<")) return maybeWithHTML.trim();
+            if (!maybeWithHTML.includes("<"))
```
Comment (simonward) [inline src/module/system/check/dialog.ts:36]: Might be better if these used localizable format strings rather than a concat.

Comment (DocSchlock) [reply src/module/system/check/dialog.ts:36]: Done!

File: static/lang/en.json [hunk 2]
```diff
@@ -3066,6 +3066,7 @@
             "Dialog": {
                 "Check": {
                     "Substitution": "Substitution",
+                    "Title": "{actorName} - {titleText}",
```
Comment (stwlam) [inline static/lang/en.json:3069]:
```suggestion
                    "Title": "{title}: {actor}",
```

File: src/module/system/check/dialog.ts [hunk 3]
```diff
@@ -29,14 +29,28 @@ export class CheckModifiersDialog extends Application {
...
 
             const div = document.createElement("div");
             div.innerHTML = maybeWithHTML;
             div.querySelector(".action-glyph, .pf2-icon")?.remove();
 
-            return div.innerText.trim();
+            return actorName
+                ? game.i18n.format("PF2E.Roll.Dialog.Check.Title", {
+                      actorName,
+                      titleText: div.innerText.trim(),
```
Comment (stwlam) [inline src/module/system/check/dialog.ts:51]:
```suggestion
                      title: div.innerText.trim(),
                      actor: actorName,
```
Bringing it in line with similar dialogs

**Comment by DocSchlock** [comment]
I changed this to show the name only on initiative, perception, and savings throws, as those would be the only roll types where a GM would potentially need this information.

**Pull request closed without merging**